context/kuberetes_client.go

```diff
@@ -0,0 +1,43 @@
+package context
+
+import (
+	"fmt"
+	"time"
+
+	dutyKubernetes "github.com/flanksource/duty/kubernetes"
+)
+
+type KubernetesClient struct {
+	*dutyKubernetes.Client
+	Connection KubernetesConnection
+	expiry     time.Time
+}
+
+func (c *KubernetesClient) SetExpiry(d time.Duration) {
+	c.expiry = time.Now().Add(d)
+}
+
+func (c *KubernetesClient) RefreshWithExpiry(ctx Context, d time.Duration) error {
+	if !c.HasExpired() {
+		return nil
+	}
+	_, rc, err := c.Connection.Populate(ctx, true)
+	if err != nil {
+		return fmt.Errorf("error refreshing kubernetes client: %w", err)
+	}
+
+	// Update rest config in place for easy reuse
+	c.Config.Host = rc.Host
+	c.Config.TLSClientConfig = rc.TLSClientConfig
+	c.Config.BearerToken = rc.BearerToken
```

Comment:
We should probably try and decode this as a JWT token and use the expiry date defined. 

If reading a kubeconfig from a secret then it could be usung BearerTokenFile / Username / Password etc so we need to handle that case.

Reply:
> If reading a kubeconfig from a secret then it could be usung BearerTokenFile / Username / Password etc so we need to handle that case.
Updated

> We should probably try and decode this as a JWT token and use the expiry date defined.
So just try to decode and see if it has an expiry ? It won't always be a jwt right?